src/Ng/Jobs/BackupV3Storage.cs

```diff
@@ -0,0 +1,236 @@
+using Microsoft.WindowsAzure.Storage.Blob;
+using NuGet.Services.Configuration;
+
+namespace Ng.Jobs
+{
+    public class BackupV3Storage : LoopingNgJob
+    {
+        private readonly string AzCopyPath = @"binaries\azcopy\azCopy.exe";
+        private readonly int DefaultBackupDays = 14;
+
+        private string _luceneStorageAccountName;
+        private string _luceneStorageKeyValue;
+        private string _luceneStorageContainer;
+        private string _catalogStorageAccountName;
+        private string _catalogStorageKeyValue;
+        private string _catalogStorageContainer;
+        private string _registrationStorageAccountName;
+        private string _registrationStorageKeyValue;
+        private string _registrationStorageContainer;
+        private string _registrationgzStorageContainer;
+        private string _destStorageAccountName;
+        private string _destStorageKeyValue;
+        private int _backupDays;
+
+        public BackupV3Storage(ILoggerFactory loggerFactory) : base(loggerFactory)
+        {
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng backupv3storage "
+                   + $"[-{Arguments.LuceneStorageAccountName} <lucene-azure-acc> "
+                   + $"-{Arguments.LuceneStorageKeyValue} <lucene-azure-key> "
+                   + $"-{Arguments.LuceneStorageContainer} <lucene-azure-container>] "
+                   + $"[-{Arguments.CatalogStorageAccountName} <catalog-azure-storage-acc> "
+                   + $"-{Arguments.CatalogStorageKeyValue} <catalog-azure-storage-key> "
+                   + $"-{Arguments.CatalogStorageContainer} <catalog container>] "
+                    + $"[-{Arguments.RegistrationStorageAccountName} <registration-azure-acc> "
+                   + $"-{Arguments.RegistrationStorageKeyValue} <registration-azure-storage-key> "
+                   + $"-{Arguments.RegistrationStorageContainer} <registration container> "
+                   + $"-{Arguments.RegistrationGZStorageContainer} <registration gz container>] "
+                   + $"[-{Arguments.DestStorageAccountName} <azure-acc> "
+                   + $"-{Arguments.DestStorageKeyValue} <azure-key> "
+                   + $"-{Arguments.DestStorageContainer} <azure-container>] "
+                   + $"-{Arguments.BackupDays} <backup Days>] "
+                   + $"[-{Arguments.VaultName} <keyvault-name> "
+                   + $"[-{Arguments.VaultName} <keyvault-name> "
+                   + $"-{Arguments.ClientId} <keyvault-client-id> "
+                   + $"-{Arguments.CertificateThumbprint} <keyvault-certificate-thumbprint> "
+                   + $"[-{Arguments.ValidateCertificate} true|false]]";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            _luceneStorageAccountName = arguments.GetOrThrow<string>(Arguments.LuceneStorageAccountName);
```

Comment:
Looks like you need an account, key, and container for each container. Could you make the arguments more generic to allow any number of containers?